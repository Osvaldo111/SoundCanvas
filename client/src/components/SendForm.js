import React from "react";
import "../style/sendForm.css";
import { connect } from "react-redux";

/**
 * author: Osvaldo Carrillo
 * This class is designed to display the send form when the user
 * wants to send the canvas via email. This class uses the props
 * "display" to show the form accordingly. And it uses the methods
 * "handleClose()" to close the form when the button is closed. And the
 * function "handleSend" to verify when the send button is pressed.
 */
class SendForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      disableSendEmpty: true
    };
  }

  handleClose = () => {
    this.props.handleDisplay();
  };

  handleSend = () => {
    this.props.handleSend(this.state.email);
  };

  handleChange = event => {
    const email = event.target.value;
    if (email.length > 0) {
      this.setState({ disableSendEmpty: false });
    } else {
      this.setState({ disableSendEmpty: true });
    }
    this.setState({ email: email });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.handleSend();
  };

  render() {
    const { display, disableBttn, message } = this.props;
    const { email, disableSendEmpty } = this.state;
    return (
      <form
        className="overSendCanv"
        style={{
          visibility: display ? "visible" : "",
          opacity: display ? "1" : ""
        }}
        onSubmit={this.handleSubmit}
      >
        <div className="sendCardConta">
          <div className="sendCardForm">
            <div className="sendCarFormTitle">Send Canvas</div>
            <div className="sendCanvFormBdy">
              <div>
                <input
                  type="email"
                  placeholder="Email Address"
                  value={email}
                  onChange={this.handleChange}
                />
              </div>
            </div>
            <div className="ruleLnFrm"></div>
            <div className="sendCvaFrmFttr">
              <input
                type="submit"
                value="Send"
                style={{
                  opacity: disableSendEmpty || disableBttn ? "0.5" : "",
                  cursor:
                    disableSendEmpty || disableBttn ? "default" : "pointer"
                }}
                disabled={disableSendEmpty || disableBttn ? true : false}
              />
              <a onClick={this.handleClose}>Close</a>
              <div>{message}</div>
            </div>
          </div>
        </div>
      </form>
    );
  }
}

function mapStateToProps(state) {}
const mapDispatchToProps = {};
export default connect(null, mapDispatchToProps)(SendForm);
