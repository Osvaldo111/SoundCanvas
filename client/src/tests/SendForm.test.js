import React from "react";
import ReactDOM from "react-dom";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SendForm from "../components/SendForm";
import { mount, shallow } from "enzyme";
//Redux
import { Provider } from "react-redux";
import store from "../reducers/store";
import { createSerializer } from "enzyme-to-json";
import { spy } from "sinon";
configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Send fomr <SendForm />", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <SendForm />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
