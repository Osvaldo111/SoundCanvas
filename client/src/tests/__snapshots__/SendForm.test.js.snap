// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send fomr <SendForm /> renders correctly 1`] = `
<Connect(SendForm)>
  <form
    className="overSendCanv"
    onSubmit={[Function]}
    style={
      Object {
        "opacity": "",
        "visibility": "",
      }
    }
  >
    <div
      className="sendCardConta"
    >
      <div
        className="sendCardForm"
      >
        <div
          className="sendCarFormTitle"
        >
          Send Canvas
        </div>
        <div
          className="sendCanvFormBdy"
        >
          <div>
            <input
              onChange={[Function]}
              placeholder="Email Address"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className="ruleLnFrm"
        />
        <div
          className="sendCvaFrmFttr"
        >
          <input
            disabled={true}
            style={
              Object {
                "cursor": "default",
                "opacity": "0.5",
              }
            }
            type="submit"
            value="Send"
          />
          <a
            onClick={[Function]}
          >
            Close
          </a>
          <div />
        </div>
      </div>
    </div>
  </form>
</Connect(SendForm)>
`;
